lib/connections/service.go

```diff
@@ -324,6 +331,16 @@ func (s *service) handle(ctx context.Context) error {
 		rd, wr := s.limiter.getLimiters(remoteID, c, isLAN)
 
 		protoConn := protocol.NewConnection(remoteID, rd, wr, c, s.model, c, deviceCfg.Compression, s.cfg.FolderPasswords(remoteID))
+		go func() {
+			// Don't immedately redial if the connection is dropped quickly.
+			delay := time.NewTimer(closedConnectionRedialMinimumAge)
+			<-protoConn.Closed()
+			<-delay.C
```

Comment:
Could check the connection's `EstablishedAt()` and just skip the rest here if it's too young. Once we're up to waiting a minute and such we can anyway let the regular redial intervals handle it I think.

Reply:
I replaced it alltogether: The benefit was, that despite the delay, for any connection closed after the initial delay, a reconnect would happen instantaneously instead of after up to the regular redial interval (1 minute). However that indeed still delays legitimate connection drops short after establishing a connection (e.g. due to cluster-config stuff). I thus added a cooldown mechanism to the redial loop instead.